gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/processor/subscription/SubscriptionProcessor.java

```diff
@@ -166,15 +179,28 @@ private String computeCtxClientIdentifier(final MutableExecutionContext ctx, fin
 
     private String computeHashOrDefault(final String valueToHash, final String fallback) {
         try {
-            MessageDigest md = MessageDigest.getInstance("SHA-256");
-            md.update(valueToHash.getBytes());
-            byte[] digest = md.digest();
-            return Hex.toHexString(digest);
-        } catch (NoSuchAlgorithmException e) {
+            Cache<String, String> hashedValueCache = getOrCreateCache();
+            String hashedValue = hashedValueCache.get(valueToHash);
+            if (hashedValue == null) {
+                MessageDigest md = MessageDigest.getInstance("SHA-256");
+                md.update(valueToHash.getBytes());
+                byte[] digest = md.digest();
+                hashedValue = Hex.toHexString(digest);
+                hashedValueCache.put(valueToHash, hashedValue);
+            }
+            return hashedValue;
+        } catch (Exception e) {
             return fallback;
         }
     }
 
+    private Cache<String, String> getOrCreateCache() {
+        return cacheManager.getOrCreateCache(
+            REMOTE_ADDRESS_HASHES_CACHE,
+            CacheConfiguration.builder().timeToLiveInMs(60_000).timeToIdleInMs(60_000).maxSize(1000).distributed(false).build()
```

Comment:
1 min is quite short. I understand there might be a lot of request. How did you arrive to the conclusion that 1 minute is the best value. I'm not asking a new parameter.